x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -675,6 +680,21 @@ private static Query translateQuery(BinaryComparison bc) {
                 return new RangeQuery(source, name, value, true, null, false, format);
             }
             if (bc instanceof LessThan) {
+                if (bc.left() instanceof StDistance && value instanceof Number) {
```

Comment:
Do we expect more similar functions to be optimized in a similar fashion? If the answer is yes, then likely we could have more infrastructure to help identify such patterns.

Reply:
I don't think we will. There will be some cases for shapes, but I don't see the similarity clearly at the moment to create a good abstraction at the moment. I would rather postpone this until we get there and then try to wrap it.